### DISCUSSION TITLE
How to disable no matching proposal found, trying alternative config function for strongswan

### DISCUSSION THREAD

**Discussion by TT2LoL** [strongswan/strongswan#2356] [Q&A]
I want procedure a no proposal chosen scenario for my testing. Then I use a not supported proposal in swanctl.conf in my linux server, but can't send any no proposal chosen message from server, fetch the swanctl log show after proposal not matching, server using a 
alternative config for it, then the corrected proposal used.

log:
IKE:AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384
12[CFG] configured proposals: IKE:AES_GCM_16_128/PRF_HMAC_SHA2_384/ECP_384
12[IKE] no matching proposal found, trying alternative config
12[CFG] selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384
![46fc22be-d348-4fe2-90b5-f4bef69fb5cc](https://github.com/user-attachments/assets/5b14c91d-db00-4931-8a7b-7cf0c515df39)

My config is : 
strongSwan swanctl 5.9.1
server: debian12

My question is how to disable the function, thanks a lot.

**Comment by tobiasbrunner** [comment]
You can't and I don't see why you would want to disable it. My guess is that you are misunderstanding something. Check your config.

**Reply by TT2LoL** [reply]
Thanks for reply, I want to check if my device could do correct process when received NO_PROPOSAL_CHOSEN  IKE_SA_INIT  response.

If it's can't disable the function in config files, I'll try other method to trigger this issue.  Thanks a lot.

**Reply by tobiasbrunner** [reply]
There will be a `NO_PROPOSAL_CHOSEN` if no connection's proposal matches, but you apparently have another connection entry that does match. So again, check your config.

**Reply by TT2LoL** [reply]
I use below swanctl.conf config, after recieved  "aes256gcm16-prfsha384-ecp384", then it's will report not matching with proposals = "aes128gcm16-prfsha384-ecp384".  Then use alternative config, the right one "aes256gcm16-prfsha384-ecp384" will be used.

I have removed all others swanctl.conf in my pc, but not worked as expect, so is there other config could cause this?

connections {
    ipsec-con-0 {
        version = 2
        local_addrs = 192.168.252.200
        remote_addrs = 192.168.252.69
        local {
            auth = pubkey
            certs = pc.ci_2029.pem
        }
        remote {
            auth = pubkey
        }
        children {
            ipsec-con-0 {
                replay_window = 64
                esp_proposals = aes256gcm16
                life_time = 2h
                rekey_time = 108m
                rand_time = 12m
                dpd_action = restart
            }
        }
        proposals = aes128gcm16-prfsha384-ecp384
        over_time = 3h
        rekey_time = 171m
        rand_time = 9m
        keyingtries = 0
    }
    passthrough-1 {
        children {
            icmp {
                local_ts = 192.168.252.200[icmp]
                remote_ts = 192.168.252.69[icmp]
                priority = 1
                mode = pass
                start_action = trap
            }
            ssh-in {
                local_ts = 192.168.252.200[tcp/ssh]
                remote_ts = 192.168.252.69[tcp]
                priority = 1
                mode = pass
                start_action = trap
            }
            ssh-out {
                local_ts = 192.168.252.200[tcp]
                remote_ts = 192.168.252.69[tcp/ssh]
                priority = 1
                mode = pass
                start_action = trap
            }
            dhcp {
                local_ts = 192.168.252.200[udp/67]
                remote_ts = 192.168.252.69[udp/68]
                priority = 1
                mode = pass
                start_action = trap
            }
        }
    }
}
charon {
    load_modular = yes
    plugins {
        include strongswan/strongswan.d/charon/*.conf
    }
}

**Reply by tobiasbrunner** [reply]
> I have removed all others swanctl.conf in my pc, but not worked as expect, so is there other config could cause this?

Yes, `passthrough-1` with all the passthrough policies (see e.g. the recent discussion in #2354 or [the docs](https://docs.strongswan.org/docs/5.9/config/quickstart.html#_passthroughbypass_policies)).

**Reply by TT2LoL** [reply]
Thanks a lot, it's working as expect after remove the passthrough-1 parameter.
